### PR TITLE
Added caseField definitions, classes for Document, DocumentWithDescription and tests

### PR DESCRIPTION
**Pull request by Ryan-Nesbitt**
### Change description ###

Added my caseField definitions for my screens, including classes for Document, DocumentWithDescription and the corresponding unit tests.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/entities/BailCaseFieldDefinition.java [hunk 1]
```diff
@@ -8,8 +8,22 @@
...
+        "applicantGivenNames", new TypeReference<String>() {}),
+    APPLICANT_FAMILY_NAME(
+        "applicantFamilyName", new TypeReference<String>() {}),
+    APPLICANT_DOB(
+        "applicantDateOfBirth", new TypeReference<String>() {}),
+    APPLICANT_GENDER(
+        "applicantGender", new TypeReference<String>() {}),
+    APPLICANT_GENDER_OTHER(
+        "applicantGenderEnterDetails", new TypeReference<String>() {}),
+    APPLICANT_NATIONALITIES(
```
Comment (neha-aggarw) [inline src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/entities/BailCaseFieldDefinition.java:25]: There is one more field to capture whether the applicant has a nationality or is stateless. Can you please check and add that here.

File: src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/entities/BailCaseFieldDefinition.java [hunk 2]
```diff
@@ -173,7 +187,17 @@
...
         "supporter4Passport", new TypeReference<String>(){}),
     FINANCIAL_AMOUNT_SUPPORTER_4_UNDERTAKES(
-        "financialAmountSupporter4Undertakes", new TypeReference<String>(){})
+        "financialAmountSupporter4Undertakes", new TypeReference<String>(){}),
+    GROUNDS_FOR_BAIL_REASONS(
+        "groundsForBailReasons", new TypeReference<String>(){}),
+    GROUNDS_FOR_BAIL_PROVIDE_EVIDENCE_OPTION(
+        "groundsForBailProvideEvidenceOption", new TypeReference<YesOrNo>(){}),
+    UPLOAD_BAIL_EVIDENCE_DOCUMENTS(
+        "uploadTheBailEvidenceDocs", new TypeReference<List<IdValue<DocumentWithDescription>>>(){}),
```
Comment (neha-aggarw) [inline src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/entities/BailCaseFieldDefinition.java:196]: I think it should be a list of DocumentWithMetaData if we want to capture the details like Date.

**Review by neha-aggarw — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/entities/BailCaseFieldDefinition.java (hunk 1 above)
Comment (Ryan-Nesbitt) [reply src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/entities/BailCaseFieldDefinition.java:27]: Added the missing APPLICANT_NATIONALITY field

File: src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/entities/BailCaseFieldDefinition.java (hunk 2 above)
Comment (Ryan-Nesbitt) [reply src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/entities/BailCaseFieldDefinition.java]: TypeReference changed to DocumentWithMetadata and added in those classes+tests

**Review by neha-aggarw — APPROVED** [review]
LGTM

**Review by rajeshthuraiyur — APPROVED** [review]
LGTM

**Pull request merged**